langwatch/src/server/api/utils.ts

```diff
@@ -34,3 +39,90 @@ export const flattenObjectKeys = (
     }
   }, []);
 };
+
+
+export async function getProtectionsForProject(
+  prisma: PrismaClient,
+  { projectId }: { projectId: string } & Record<string, unknown>
+): Promise<Protections> {
+  return await getUserProtectionsForProject({ prisma, session: null, publiclyShared: false }, { projectId });
+}
+
+export async function getUserProtectionsForProject(
+  ctx: { prisma: PrismaClient; session: Session | null; publiclyShared?: boolean },
+  { projectId }: { projectId: string } & Record<string, unknown>
+): Promise<Protections> {
+  // TODO(afr): Should we show cost if public? I would assume the opposite.
+  const canSeeCosts = ctx.publiclyShared || await backendHasTeamProjectPermission(
+    ctx,
+    { projectId },
+    TeamRoleGroup.COST_VIEW
+  );
+
+  const project = await ctx.prisma.project.findUniqueOrThrow({
+    where: { id: projectId },
+    select: { 
+      capturedInputVisibility: true,
+      capturedOutputVisibility: true,
+    },
+  });
+
+  // For public shares or non-signed in users, we only check project settings
+  if (ctx.publiclyShared || !ctx.session?.user?.id) {
+    return {
+      canSeeCosts,
+      canSeeCapturedInput: project.capturedInputVisibility === ProjectSensitiveDataVisibilityLevel.VISIBLE_TO_ALL,
+      canSeeCapturedOutput: project.capturedOutputVisibility === ProjectSensitiveDataVisibilityLevel.VISIBLE_TO_ALL,
+    };
+  }
+
+  // For signed in users, check their team permissions
+  const teamsWithAccess = await ctx.prisma.teamUser.findMany({
+    where: {
+      userId: ctx.session.user.id,
+      team: {
+        projects: {
+          some: {
+            id: projectId
+          }
+        }
+      }
+    },
+    select: {
+      role: true
+    },
+  });
+
+  const isAdminInAnyTeam = teamsWithAccess.some(team => team.role === TeamUserRole.ADMIN);
+  const isMemberInAnyTeam = teamsWithAccess.length > 0;
+
+  const canAccessSensitiveData = (
+    visibility: ProjectSensitiveDataVisibilityLevel,
+    userAccess: { isAdmin: boolean; isMember: boolean }
+  ): boolean => {
+    switch (true) {
+      case !userAccess.isMember:
+        return false;
+      case visibility === ProjectSensitiveDataVisibilityLevel.REDACTED_TO_ALL:
+        return false;
+      case visibility === ProjectSensitiveDataVisibilityLevel.VISIBLE_TO_ALL:
+        return true;
+      case visibility === ProjectSensitiveDataVisibilityLevel.VISIBLE_TO_ADMIN:
+        return userAccess.isAdmin;
+      default:
+        console.error('Unexpected state for visibility:', visibility);
+        return false;
+    }
+  };
+
+  const userAccess = {
+    isAdmin: isAdminInAnyTeam,
+    isMember: isMemberInAnyTeam,
+  };
+
+  return {
+    canSeeCosts,
+    canSeeCapturedInput: canAccessSensitiveData(project.capturedInputVisibility, userAccess),
+    canSeeCapturedOutput: canAccessSensitiveData(project.capturedOutputVisibility, userAccess),
```

Comment:
it's weird that you're mixing values and functions here with the same naming. 
Also, it seems weird that you'd return the function as part of a payload object, rather than simply exposing the methods as part of the API.